### ISSUE TITLE
Async class method is not recognized

### ISSUE THREAD

**Issue by q4w56** [eslint/eslint#9674]
**Tell us about your environment**

* **ESLint Version:** 4.12.1
* **Node Version:** 8.9.0
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "env": {
        "browser": true,
    },
    "extends": ["eslint:recommended"],
    "parserOptions": {
        "ecmaVersion": 7,
    },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// main.js
class Duck {
  async notPassed(){}
}

async function passed(){

}
```

```bash
eslint main.js
```

**What did you expect to happen?**
No error
**What actually happened? Please include the actual, raw output from ESLint.**
```
/home/mmmmmm/main.js
  2:8  error  Parsing error: Unexpected token notPassed

✖ 1 problem (1 error, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
Closing as a duplicate of https://github.com/eslint/eslint/issues/9572
